cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -535,10 +535,19 @@ data ProjectPackageLocation =
 -- | Exception thrown by 'findProjectPackages'.
 --
 newtype BadPackageLocations = BadPackageLocations [BadPackageLocation]
+#if MIN_VERSION_base(4,8,0)
   deriving (Show, Typeable)
+#else
+  deriving (Typeable)
 
-instance Exception BadPackageLocations
---TODO: [required eventually] displayException for nice rendering
+instance Show BadPackageLocations where
```

Comment:
I don't like this Show instance; Show should print out a Haskell style message. Maybe we should add an Outputable type class for pretty-printable entities.